### PR TITLE
Made the generator platform agnostic

### PR DESCRIPTION
**Pull request by saichand345**
This generator can now be used in windows as well for creating,building and deploying service fabric applications.

### CODE REVIEW

File: generators/CoreCLRStatefulActor/index.js [hunk 1]
```diff
@@ -85,8 +85,8 @@ var ClassGenerator = generators.Base.extend({
  
     var appPackagePath = this.isAddNewService == false ? path.join(this.props.projName, appPackage) :  appPackage;
     var serviceSrcPath = this.isAddNewService == false ? path.join(this.props.projName, serviceProjName) : path.join(this.props.projName, serviceProjName) ;
-    var interfaceSrcPath = this.isAddNewService == false ? path.join(this.props.projName, interfaceProjName) : interfaceProjName;
-    var testClientSrcPath = this.isAddNewService == false ? path.join(this.props.projName, testClientProjName) : testClientProjName;
+    var interfaceSrcPath = this.isAddNewService == false ? path.join(this.props.projName, interfaceProjName) : path.join(this.props.projName, interfaceProjName);
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/index.js]: Am I missing something here? Both the cases looks same here.. Do we need the ternary operator?

File: generators/CoreCLRStatefulActor/index.js [hunk 2]
```diff
@@ -95,16 +95,36 @@ var ClassGenerator = generators.Base.extend({
...
     var testCodePath = path.join(appPackage , serviceProjName + 'TestClient' );
 
+    var is_Windows = (process.platform=='win32');
+    var is_Linux = (process.platform=='linux');
+    var is_mac = (process.platform=='darwin');
+
+    var extension1;
+    var extension2;
+    var serviceManifestFile;
+    if(is_Windows)
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/index.js]: Nit Pik: Add space after if

File: generators/CoreCLRStatefulActor/index.js [hunk 3]
```diff
@@ -95,16 +95,36 @@ var ClassGenerator = generators.Base.extend({
...
+    {
+      extension1 = '.ps1';
+      extension2 = '.cmd';
+      serviceManifestFile = 'ServiceManifest.xml';
+    }
+    else if(is_Linux){
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/index.js]: What about MAC?

File: generators/CoreCLRStatefulActor/index.js [hunk 4]
```diff
@@ -95,16 +95,36 @@ var ClassGenerator = generators.Base.extend({
     var codePath = path.join(appPackage , appPackagePath, servicePackage, 'Code');
     var testCodePath = path.join(appPackage , serviceProjName + 'TestClient' );
 
+    var is_Windows = (process.platform=='win32');
+    var is_Linux = (process.platform=='linux');
+    var is_mac = (process.platform=='darwin');
+
+    var extension1;
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/index.js]: Better to name this as sdkScriptExtension and entryPointScriptExtension

File: generators/CoreCLRStatefulActor/index.js [hunk 5]
```diff
@@ -140,18 +160,20 @@ var ClassGenerator = generators.Base.extend({
     }
 
     this.fs.copyTpl(
-      this.templatePath('service/app/appPackage/servicePackage/Code/entryPoint.sh'),
-      this.destinationPath(path.join(appPackage , appPackagePath, servicePackage, 'Code', 'entryPoint.sh')),
+      this.templatePath('service/app/appPackage/servicePackage/Code/entryPoint'+extension2),
+      this.destinationPath(path.join(appPackage , appPackagePath, servicePackage, 'Code', 'entryPoint'+extension2)),
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/index.js]: Are we still using entrypoint script for windows?

File: generators/CoreCLRStatefulActor/index.js [hunk 6]
```diff
@@ -140,18 +160,20 @@ var ClassGenerator = generators.Base.extend({
...
         serviceProjName : serviceProjName
       } 
     );
-    this.fs.copyTpl(
-      this.templatePath('main/common/dotnet-include.sh'),
-      this.destinationPath(path.join(appPackage, appPackagePath, servicePackage, 'Code', 'dotnet-include.sh')),
-      {
-      }
-    );
+    if(is_Linux){
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/index.js]: Mac?

File: generators/CoreCLRStatefulActor/index.js [hunk 7]
```diff
@@ -266,33 +288,44 @@ var ClassGenerator = generators.Base.extend({
...
         {
           appPackage: appPackage,
           appName: appName,
           appTypeName: appTypeName
         }
       );
+
+      if(is_Windows){
+        this.fs.copyTpl(
+          this.templatePath('main/deploy/predeploy'+ extension1),
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/index.js]: What is this?

File: generators/CoreCLRStatefulActor/templates/main/deploy/predeploy.ps1 [hunk 8]
```diff
@@ -0,0 +1,3 @@
+$sdkInstallPath = (Get-ItemProperty 'HKLM:\Software\Microsoft\Service Fabric SDK').FabricSDKInstallPath
+$sfSdkPsModulePath = $sdkInstallPath + "Tools\PSModule\ServiceFabricSDK"
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/templates/main/deploy/predeploy.ps1]: This shouldn't be needed I thought.

File: generators/CoreCLRStatefulActor/templates/service/app/appPackage/servicePackage/Code/entryPoint.cmd [hunk 9]
```diff
@@ -0,0 +1,2 @@
+dotnet %~dp0\<%= serviceProjName %>.dll
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/templates/service/app/appPackage/servicePackage/Code/entryPoint.cmd]: Let's get rid of entrypoint in Windows . I thought we discussed this

File: generators/CoreCLRStatefulService/index.js [hunk 10]
```diff
@@ -81,13 +81,33 @@ var ClassGenerator = generators.Base.extend({
     var serviceProject = path.join(appPackage , 'src' , serviceSrcPath , serviceProjName + '.csproj');
     var codePath = path.join(appPackage , appPackagePath, servicePackage, 'Code');
 
+    var is_Windows = (process.platform=='win32');
+    var is_Linux = (process.platform=='linux');
+    var is_mac = (process.platform=='darwin');
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulService/index.js]: Same comments as above for other types as well..

File: generators/CoreCLRStatefulActor/index.js (hunk 1 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/index.js]: We dont need it. I made ad-hoc changes and forgot to remove them. Now removed.

File: generators/CoreCLRStatefulActor/index.js (hunk 2 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/index.js]: Done

File: generators/CoreCLRStatefulActor/index.js (hunk 4 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/index.js]: Done named them as sdkScriptExtension and buildScriptExtension as we are not using entrypoint script for windows any more.

File: generators/CoreCLRStatefulActor/index.js (hunk 5 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/index.js]: No we are not using. We were generating it but not actually using it . Now its removed.

File: generators/CoreCLRStatefulActor/index.js (hunk 7 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/index.js]: This was supposed to be used for loading powershell scripts before using service fabric commandlets. But apparently we no more need this and hence this is removed.

File: generators/CoreCLRStatefulActor/templates/main/deploy/predeploy.ps1 (hunk 8 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/templates/main/deploy/predeploy.ps1]: Yeah we dont need it . Removed

File: generators/CoreCLRStatefulActor/templates/service/app/appPackage/servicePackage/Code/entryPoint.cmd (hunk 9 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/templates/service/app/appPackage/servicePackage/Code/entryPoint.cmd]: Yeah we are not using entrypoint script for windows. We were generating it but actuallly not using it.Now its removed.

File: generators/CoreCLRStatefulService/index.js (hunk 10 above)
Comment (saichand345) [reply generators/CoreCLRStatefulService/index.js]: Done.

File: generators/CoreCLRStatefulActor/index.js (hunk 3 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/index.js]: Added support for MAC and it has servicemanifest same as that of windows (avoiding entrypoint)

File: generators/CoreCLRStatefulActor/index.js (hunk 6 above)
Comment (saichand345) [reply generators/CoreCLRStatefulActor/index.js]: We don't need entrypoint for MAC as discussed.

File: generators/CoreCLRStatefulActor/index.js [hunk 11]
```diff
@@ -95,16 +95,37 @@ var ClassGenerator = generators.Base.extend({
     var codePath = path.join(appPackage , appPackagePath, servicePackage, 'Code');
     var testCodePath = path.join(appPackage , serviceProjName + 'TestClient' );
 
+    var is_Windows = (process.platform == 'win32');
+    var is_Linux = (process.platform == 'linux');
+    var is_mac = (process.platform == 'darwin');
```
Comment (rapatchi) [inline generators/CoreCLRStatefulActor/index.js]: Test this on mac for all generators

**Pull request merged**